sympy/calculus/util.py

```diff
@@ -947,13 +947,23 @@ class AccumulationBounds(AtomicExpr):
     >>> 1/AccumBounds(-oo, 0)
     AccumBounds(-oo, 0)
 
-    A boundary of 1 will always generate all nonnegatives:
+    A boundary of 1.0 will always generate all nonnegatives since
+    the 1.0 represents a number with some uncertainty so it is
+    always rounded in a direction to keep the returned bounds as
+    generous as possible:
 
-    >>> AccumBounds(1, 2)**oo
+    >>> AccumBounds(1.0, 2)**oo
     AccumBounds(0, oo)
-    >>> AccumBounds(0, 1)**oo
+    >>> AccumBounds(0, 1.0)**oo
     AccumBounds(0, oo)
 
+    A boundary of 1 will be assumed as exact:
+
+    >>> AccumBounds(1, 2)**oo
+    AccumBounds(1, oo)
+    >>> AccumBounds(0, 1)**oo
+    AccumBounds(0, 1)
```

Comment:
AccumulationBounds denotes an interval containing the limits determined by some expression. This example would imply that if `f` has limit in the interval [0, 1] and `g` has limit `oo`, then also `f**g` would have limit in the same interval. But that is not true, e.g. for `f = 1 + 1/x, g = x, x -> oo`.

Reply:
What does sympy give for the limit of `(1 +1/x)**x`? I think we have to avoid putting possible history on states or else we can't resolve cases that are resolvable. The limit of `sin(x)**x` must lie between -1 and 1 since it strictly never exceeds an absolute value of 1. 0.0 is a better answer since you end up with delta functions at the max/min locations and 0 everywhere else.

If a bound approaches 1 from the positive side then a bound should use 1.0 and the behavior should be right. At least that is my goal in adding this nuance.

Reply:
I guess `AccumBounds` represents a limit so it has to be handled differently from ordinary interval arithmetic if the result depends on which side a limit would approach the boundary.

Reply:
The question for me is, "Does allowing AccumBounds to differentiate between exact and inexact bounds break anything?" If you say that this nuance is just a bad idea for people using AccumBounds as a limit-related quantity used for reasoning then I will trust you. But if the master behavior of `(sin(x)**x).subs(x,oo)` giving `(-oo,oo)` as the result indicates to me that it is not a very powerful reasoning tool at present.

I guess what I am looking for are demonstrations of how this refinement will hurt reasoning with AccumBounds and SymPy.

Here are some current behaviors using @jksuom example:
```python
>>> (1+1/x)**x
(1 + 1/x)**x
>>> eq=_;limit(eq,x,oo)
E
>>> eq.subs(x,oo)
nan
>>> (1+1/x).subs(x,oo)
1.00000000000000
>>> _**oo
nan
```
Hmm...I tried one more and note something interesting:
```python
>>> (1-1/x).subs(x,oo)
1
>>> ((1-1/x)**x).subs(x,oo)
1
```
This is the result of the difference
```
>>> 1 + 0.
1.0
>>> 1 - 0.
1
```

Reply:
> ((1-1/x)**x).subs(x,oo)

`subs` can sometimes be used as a poor man's limit but one must be cautious. The correct limit value is `1/E`.

Reply:
> The limit of sin(x)**x must lie between -1 and 1

This is not a good example as `sin(x)**x` is not real when `sin(x)` is negative, hence not int the interval between -1 and 1. In this case, there is no object in SymPy that would describe the limit better than `nan`. It is not optimal as there is no size information but we don't have complex analogs of `AccumBounds`.

Reply:
> Does allowing AccumBounds to differentiate between exact and inexact bounds break anything?

I would generally try to avoid floating point numbers in SymPy but AccumBounds should behave in the same way even if the bounds are Floats.

Reply:
I'll see if I can patch around the changes outside of AccumBounds that affect AccumBounds (my `1oob` branch).

Reply:
> AccumBounds should behave in the same way even if the bounds are Floats.

This is where we then end up with inconsistent behavior like `1**x == 1` but `1.0**x -> unevaluated`. We might be leaving the latter unevaluated because we don't know if it goes to 0 or infinity if x is infinite (though it might just be a quirk since `1.0**I` also remains unevaluated).